keps/sig-node/3721-support-for-env-files/README.md

```diff
@@ -0,0 +1,565 @@
+and `FileKeySelector` field under [`EnvVarSource`](https://pkg.go.dev/k8s.io/api/core/v1#EnvVarSource)
+that can be set for an individual container.
+
+### Pod API
+
+`FileEnvSource` field will allow the users to populate environment variables
+from the specified file. The user can specify a file using `VolumeName` and
+`Path` fields. The `VolumeName` field specifies the volume mount that contains
+the file and `Path` is the relative path in this volume mount filesystem.
+
+```
+type EnvFromSource struct {
+    ...
+    // The file to select from
+    // +optional
+    FileRef *FileEnvSource `json:"fileRef,omitempty" protobuf:"bytes,3,opt,name=fileRef"`
+    ...
+}
+
+type FileEnvSource struct {
+    // The name of the volume mount containing the env file.
+    VolumeName string `json:",inline" protobuf:"bytes,1,opt,name=volumeName"`
+    // The relative file path inside the volume mount to select from.
+    Path string `json:",inline" protobuf:"bytes,2,opt,name=path"`
+    // Specify whether the file must exist. If the file does not exist,
+    // then the env var is not published.
+    // +optional
+    Optional *bool `json:"optional,omitempty" protobuf:"varint,3,opt,name=optional"`
+}
+```
+
+`FileKeySelector` field will allow the users to select the value for environment
+variable from the specified key in the file.
+
+```
+type EnvVarSource struct {
+    ...
+    // Selects a key of the env file.
+    // +optional
+    FileKeyRef *FileKeySelector `json:"fileKeyRef,omitempty" protobuf:"bytes,5,opt,name=fileKeyRef"`
+    ...
+}
+
+type FileKeySelector struct {
+    // The name of the volume mount containing the env file.
+    VolumeName string `json:",inline" protobuf:"bytes,1,opt,name=volumeName"`
+    // The relative file path inside the volume mount to select from.
+    Path string `json:",inline" protobuf:"bytes,2,opt,name=path"`
+    // The key of the env file to select from.  Must be a valid key.
+    Key string `json:"key" protobuf:"bytes,3,opt,name=key"`
+    // Specify whether the file or its key must be defined. If the file or key
+    // does not exist, then the env var is not published.
+    // +optional
+    Optional *bool `json:"optional,omitempty" protobuf:"varint,4,opt,name=optional"`
+}
+```
+
+### Env File
+
+The full specification of an env file:
+
+1. **File Format**: The environment variable (env) file must adhere to valid [.env syntax](https://smartmob-rfc.readthedocs.io/en/latest/2-dotenv.html) to ensure correct parsing. An example:
+
+```
+KEY1=VALUE1
+KEY2=VALUE2
+...
+```
+
+2. **Variable Naming**:
+    
+    a. **Standard**: Environment variable names should be composed of alphanumeric characters (letters and numbers), underscores, dots, or hyphens. The first character cannot be a digit.
+
+    b. **Relaxed**: If the Kubernetes feature gate `RelaxedEnvironmentVariableValidation` is enabled, the naming restriction is less stringent, allowing any printable ASCII character except the equals sign (=).
```

Comment:
I think this can be more specifically just: apply the same variable name restrictions as other API-defined env vars (i.e. https://github.com/kubernetes/kubernetes/blob/a7ca13ea29ba5b3c91fd293cdbaec8fb5b30cee2/pkg/apis/core/validation/validation.go#L2583-L2596)

Reply:
Done.